lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -135,20 +136,35 @@ export const migrateFileRecord = async (
 
 /**
  * Migrate granule and files from DynamoDB to RDS
- * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
- * @param {GranulesAndFilesMigrationSummary} granuleAndFileMigrationSummary
- * @param {Knex} knex
- * @param {number} loggingInterval
+ * @param {Object} params
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} params.dynamoRecord
+ * @param {GranulesAndFilesMigrationSummary} params.granuleAndFileMigrationSummary
+ * @param {Knex} params.knex
+ * @param {number} params.loggingInterval
+ * @param {string} params.bucket
+ * @param {string} params.stackName
  * @returns {Promise<MigrationSummary>} - Migration summary for granules and files
  */
-export const migrateGranuleAndFilesViaTransaction = async (
+export const migrateGranuleAndFilesViaTransaction = async (params: {
   dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
   granuleAndFileMigrationSummary: GranulesAndFilesMigrationSummary,
   knex: Knex,
-  loggingInterval: number
-): Promise<GranulesAndFilesMigrationSummary> => {
+  loggingInterval: number,
+  bucket: string
+  stackName: string,
+}): Promise<GranulesAndFilesMigrationSummary> => {
+  const {
+    dynamoRecord,
+    granuleAndFileMigrationSummary,
+    knex,
+    loggingInterval,
+    bucket,
+    stackName,
+  } = params;
   const files = dynamoRecord.files;
   const { granulesSummary, filesSummary } = granuleAndFileMigrationSummary;
+  const errorFile = [];
+  let errorMessage;
```

Comment:
This doesn't appear to be used anywhere but in the catch below, consider moving it to around L182 where it's used rather than defining it here.